Repeated rebuilding of compiler

1. Select a test in Test Explorer
2. Run it
3. FSharp.Compiler.Private builds
4. Change no code, rerun the test
5. FSharp.Compiler.Private builds again

Investigating the cause with this registry flag https://blogs.msdn.microsoft.com/kirillosenkov/2014/08/04/how-to-investigate-rebuilding-in-visual-studio-when-nothing-has-changed/ I get the following output:
```
2>Checking whether FSharp.Compiler.Private needs to be rebuilt:
2>   Output: 15-Sep-17 16:00:20 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSHARP.COMPILER.PRIVATE\OBJ\NET40\FSHARP.COMPILER.PRIVATE.DLL
2>   Output: 03-Aug-16 14:18:52 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\MICROSOFT.BUILD.FRAMEWORK.DLL
2>   Output: 03-Aug-16 14:18:52 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\MICROSOFT.BUILD.DLL
2>   Output: 03-Aug-16 14:18:52 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\MICROSOFT.BUILD.CONVERSION.CORE.DLL
2>   Output: 03-Aug-16 14:18:52 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\MICROSOFT.BUILD.ENGINE.DLL
2>   Output: 03-Aug-16 14:18:52 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\MICROSOFT.BUILD.UTILITIES.CORE.DLL
2>   Output: 03-Aug-16 14:18:52 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\MICROSOFT.BUILD.TASKS.CORE.DLL
2>   Output: 13-Jun-16 05:30:36 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\MICROSOFT.DIASYMREADER.PORTABLEPDB.DLL
2>   Output: 19-Jan-17 16:23:52 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\MICROSOFT.DIASYMREADER.DLL
2>   Output: 19-Dec-16 12:20:32 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\SYSTEM.REFLECTION.METADATA.DLL
2>   Output: 11-Jun-16 23:13:16 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\SYSTEM.COLLECTIONS.IMMUTABLE.DLL
2>   Output: 19-Apr-17 16:58:52 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\SYSTEM.VALUETUPLE.DLL
2>   Output: 15-Sep-17 15:59:47 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\FSHARP.COMPILER.PRIVATE.XML
2>   Output: 15-Sep-17 16:00:20 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSHARP.COMPILER.PRIVATE\OBJ\NET40\FSHARP.COMPILER.PRIVATE.PDB
2>   Output: 15-Sep-17 16:00:20 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\FSHARP.COMPILER.PRIVATE.DLL
2>   Output: 15-Sep-17 16:00:20 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\FSHARP.COMPILER.PRIVATE.PDB
2>   Input: 15-Sep-17 11:26:57 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSHARP.COMPILER.PRIVATE\FSHARP.COMPILER.PRIVATE.FSPROJ
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ASSEMBLYINFO\ASSEMBLYINFO.FSHARP.COMPILER.PRIVATE.DLL.FS
2>   Input: 04-Sep-17 15:49:52 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\RESHAPEDREFLECTION.FS
2>   Input: 04-Sep-17 15:49:52 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\RESHAPEDMSBUILD.FS
2>   Input: 15-Sep-17 11:26:48 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\SFORMAT.FSI
2>   Input: 15-Sep-17 11:26:48 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\SFORMAT.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\SR.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\SR.FS
2>   Input: 04-Sep-17 15:49:52 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\PRIM-LEXING.FSI
2>   Input: 04-Sep-17 15:49:52 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\PRIM-LEXING.FS
2>   Input: 04-Sep-17 15:49:52 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\PRIM-PARSING.FSI
2>   Input: 04-Sep-17 15:49:52 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\PRIM-PARSING.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\RESIZEARRAY.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\RESIZEARRAY.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\HASHMULTIMAP.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\HASHMULTIMAP.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\EDITDISTANCE.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\TAGGEDCOLLECTIONS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\TAGGEDCOLLECTIONS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILDIAG.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILDIAG.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILLIB.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\FILENAME.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\FILENAME.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ZMAP.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ZMAP.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ZSET.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ZSET.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\BYTES.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\BYTES.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\INTERNALCOLLECTIONS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\INTERNALCOLLECTIONS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\QUEUELIST.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\LIB.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\RATIONAL.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\RATIONAL.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\RANGE.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\RANGE.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\ERRORLOGGER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\ERRORRESOLUTIONHINTS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\REFERENCERESOLVER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\IL.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\IL.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILX.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILX.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILASCII.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILASCII.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILPRINT.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILPRINT.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILMORPH.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILMORPH.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILSUPP.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILSUPP.FS
2>   Input: 15-Sep-17 12:39:08 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSHARP.COMPILER.PRIVATE\ILPARS.FS
2>   Input: 15-Sep-17 12:39:06 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSHARP.COMPILER.PRIVATE\ILLEX.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILBINARY.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILBINARY.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILREAD.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILREAD.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILWRITE.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILWRITEPDB.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILWRITEPDB.FS
2>   Input: 15-Sep-17 11:26:48 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILWRITE.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ABSIL\ILREFLECT.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\UTILS\COMPILERLOCATIONUTILS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\PRETTYNAMING.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ILX\ILXSETTINGS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ILX\ERASECLOSURES.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ILX\ERASECLOSURES.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ILX\ERASEUNIONS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\ILX\ERASEUNIONS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\UNICODELEXING.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\UNICODELEXING.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\LAYOUT.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\LAYOUT.FS
2>   Input: 15-Sep-17 11:26:48 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\AST.FS
2>   Input: 15-Sep-17 12:39:08 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSHARP.COMPILER.PRIVATE\PPPARS.FS
2>   Input: 15-Sep-17 12:39:26 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSHARP.COMPILER.PRIVATE\PARS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\LEXHELP.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\LEXHELP.FS
2>   Input: 15-Sep-17 12:39:06 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSHARP.COMPILER.PRIVATE\PPLEX.FS
2>   Input: 15-Sep-17 12:39:07 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSHARP.COMPILER.PRIVATE\LEX.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\LEXFILTER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TAINTED.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TAINTED.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\EXTENSIONTYPING.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\EXTENSIONTYPING.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\QUOTATIONPICKLER.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\QUOTATIONPICKLER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TAST.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TCGLOBALS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TASTOPS.FSI
2>   Input: 15-Sep-17 11:26:48 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TASTOPS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TASTPICKLE.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TASTPICKLE.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\IMPORT.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\IMPORT.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\INFOS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\ACCESSIBILITYLOGIC.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\ATTRIBUTECHECKING.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\INFOREADER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\NICEPRINT.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\AUGMENTWITHHASHCOMPARE.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\AUGMENTWITHHASHCOMPARE.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\NAMERESOLUTION.FSI
2>   Input: 15-Sep-17 11:26:48 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\NAMERESOLUTION.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TYPERELATIONS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\SIGNATURECONFORMANCE.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\METHODOVERRIDES.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\METHODCALLS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\PATTERNMATCHCOMPILATION.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\PATTERNMATCHCOMPILATION.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\CONSTRAINTSOLVER.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\CONSTRAINTSOLVER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\CHECKFORMATSTRINGS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\CHECKFORMATSTRINGS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FINDUNSOLVED.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\QUOTATIONTRANSLATOR.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\QUOTATIONTRANSLATOR.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\POSTINFERENCECHECKS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\POSTINFERENCECHECKS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TYPECHECKER.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\TYPECHECKER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\OPTIMIZER.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\OPTIMIZER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\DETUPLEARGS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\DETUPLEARGS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\INNERLAMBDASTOTOPLEVELFUNCS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\INNERLAMBDASTOTOPLEVELFUNCS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\LOWERCALLSANDSEQS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\AUTOBOX.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\ILXGEN.FSI
2>   Input: 15-Sep-17 11:26:48 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\ILXGEN.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\COMPILEOPS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\COMPILEOPS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\COMPILEOPTIONS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\COMPILEOPTIONS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSC.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSC.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\SYMBOLS\SYMBOLHELPERS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\SYMBOLS\SYMBOLHELPERS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\SYMBOLS\SYMBOLS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\SYMBOLS\SYMBOLS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\SYMBOLS\EXPRS.FSI
2>   Input: 15-Sep-17 11:26:48 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\SYMBOLS\EXPRS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\INCREMENTALBUILD.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\INCREMENTALBUILD.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\REACTOR.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\REACTOR.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICECONSTANTS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEDECLARATIONLISTS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEDECLARATIONLISTS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICELEXING.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICELEXING.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEPARSETREEWALK.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICENAVIGATION.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICENAVIGATION.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEPARAMINFOLOCATIONS.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEPARAMINFOLOCATIONS.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEUNTYPEDPARSE.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEUNTYPEDPARSE.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEASSEMBLYCONTENT.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEASSEMBLYCONTENT.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEXMLDOCPARSER.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEXMLDOCPARSER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\SIMULATEDMSBUILDREFERENCERESOLVER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\EXTERNALSYMBOL.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\EXTERNALSYMBOL.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICE.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICE.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEINTERFACESTUBGENERATOR.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICEINTERFACESTUBGENERATOR.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICESTRUCTURE.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\VS\SERVICESTRUCTURE.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSI\FSI.FSI
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSI\FSI.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSHARP.COMPILER.PRIVATE\INTERNALSVISIBLETO.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\MSBUILDREFERENCERESOLVER.FS
2>   Input: 04-Sep-17 15:49:51 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\LEGACYHOSTEDCOMPILERFORTESTING.FS
2>   Input: 21-Jun-17 11:36:13 D:\WLCR\GITHUB\VISUALFSHARP\SRC\FSHARP\FSSTRINGS.RESX
2>   Input: 15-Sep-17 12:38:57 D:\WLCR\GITHUB\VISUALFSHARP\DEBUG\NET40\BIN\FSHARP.CORE.DLL
2>   Input: 26-Sep-12 04:16:08 C:\PROGRAM FILES (X86)\REFERENCE ASSEMBLIES\MICROSOFT\FRAMEWORK\.NETFRAMEWORK\V4.5\ISYMWRAPPER.DLL
2>   Input: 03-Aug-16 14:18:52 D:\WLCR\GITHUB\VISUALFSHARP\PACKAGES\MICROSOFT.VISUALFSHARP.MSBUILD.15.0.1.0.1\LIB\NET45\MICROSOFT.BUILD.DLL
2>   Input: 03-Aug-16 14:18:52 D:\WLCR\GITHUB\VISUALFSHARP\PACKAGES\MICROSOFT.VISUALFSHARP.MSBUILD.15.0.1.0.1\LIB\NET45\MICROSOFT.BUILD.FRAMEWORK.DLL
2>   Input: 03-Aug-16 14:18:52 D:\WLCR\GITHUB\VISUALFSHARP\PACKAGES\MICROSOFT.VISUALFSHARP.MSBUILD.15.0.1.0.1\LIB\NET45\MICROSOFT.BUILD.TASKS.CORE.DLL
2>   Input: 03-Aug-16 14:18:52 D:\WLCR\GITHUB\VISUALFSHARP\PACKAGES\MICROSOFT.VISUALFSHARP.MSBUILD.15.0.1.0.1\LIB\NET45\MICROSOFT.BUILD.UTILITIES.CORE.DLL
2>   Input: 19-Jan-17 16:23:52 D:\WLCR\GITHUB\VISUALFSHARP\PACKAGES\MICROSOFT.DIASYMREADER.1.1.0\LIB\PORTABLE-NET45+WIN8\MICROSOFT.DIASYMREADER.DLL
2>   Input: 13-Jun-16 05:30:36 D:\WLCR\GITHUB\VISUALFSHARP\PACKAGES\MICROSOFT.DIASYMREADER.PORTABLEPDB.1.1.0\LIB\PORTABLE-NET45+WIN8\MICROSOFT.DIASYMREADER.PORTABLEPDB.DLL
2>   Input: 26-Sep-12 04:16:08 C:\PROGRAM FILES (X86)\REFERENCE ASSEMBLIES\MICROSOFT\FRAMEWORK\.NETFRAMEWORK\V4.5\MSCORLIB.DLL
2>   Input: 26-Sep-12 04:16:08 C:\PROGRAM FILES (X86)\REFERENCE ASSEMBLIES\MICROSOFT\FRAMEWORK\.NETFRAMEWORK\V4.5\SYSTEM.DLL
2>   Input: 11-Jun-16 23:13:16 D:\WLCR\GITHUB\VISUALFSHARP\PACKAGES\SYSTEM.COLLECTIONS.IMMUTABLE.1.2.0\LIB\PORTABLE-NET45+WIN8+WP8+WPA81\SYSTEM.COLLECTIONS.IMMUTABLE.DLL
2>   Input: 26-Sep-12 04:16:08 C:\PROGRAM FILES (X86)\REFERENCE ASSEMBLIES\MICROSOFT\FRAMEWORK\.NETFRAMEWORK\V4.5\SYSTEM.CORE.DLL
2>   Input: 26-Sep-12 04:16:08 C:\PROGRAM FILES (X86)\REFERENCE ASSEMBLIES\MICROSOFT\FRAMEWORK\.NETFRAMEWORK\V4.5\SYSTEM.NUMERICS.DLL
2>   Input: 19-Dec-16 12:20:32 D:\WLCR\GITHUB\VISUALFSHARP\PACKAGES\SYSTEM.REFLECTION.METADATA.1.4.2\LIB\PORTABLE-NET45+WIN8\SYSTEM.REFLECTION.METADATA.DLL
2>   Input: 19-Apr-17 16:58:52 D:\WLCR\GITHUB\VISUALFSHARP\PACKAGES\SYSTEM.VALUETUPLE.4.3.1\LIB\NETSTANDARD1.0\SYSTEM.VALUETUPLE.DLL
2>Freshest input: 15-Sep-17 12:39:26
2>Stalest output: 11-Jun-16 23:13:16
```
The stalest output corresponds to a single assembly:
```
Input: 11-Jun-16 23:13:16 D:\WLCR\GITHUB\VISUALFSHARP\PACKAGES\SYSTEM.COLLECTIONS.IMMUTABLE.1.2.0\LIB\PORTABLE-NET45+WIN8+WP8+WPA81\SYSTEM.COLLECTIONS.IMMUTABLE.DLL
```
Any ideas what to do with it?

--- a/build.cmd
+++ b/build.cmd
@@ -639,7 +639,7 @@ if "%OSARCH%"=="AMD64" set SYSWOW64=SysWoW64
 
 if not "%OSARCH%"=="x86" set REGEXE32BIT=%WINDIR%\syswow64\reg.exe
 
-echo SDK environment vars from Registry
+echo SDK environment vars from Registry (note: ignore "ERROR: The system was unable to find ....")
 echo ==================================
                             FOR /F "tokens=2* delims=	 " %%A IN ('%REGEXE32BIT% QUERY "HKLM\Software\WOW6432Node\Microsoft\Microsoft SDKs\NETFXSDK\4.6.2\WinSDK-NetFx40Tools" /v InstallationFolder') DO SET WINSDKNETFXTOOLS=%%B
 if "%WINSDKNETFXTOOLS%"=="" FOR /F "tokens=2* delims=	 " %%A IN ('%REGEXE32BIT% QUERY "HKLM\Software\WOW6432Node\Microsoft\Microsoft SDKs\NETFXSDK\4.6.1\WinSDK-NetFx40Tools" /v InstallationFolder') DO SET WINSDKNETFXTOOLS=%%B

--- a/src/fsharp/FSharp.Build/FSharpEmbedResourceText.fs
+++ b/src/fsharp/FSharp.Build/FSharpEmbedResourceText.fs
@@ -341,12 +341,21 @@ open Printf
 
     let generateResxAndSource (filename:string) =
         try
-            let printMessage message = printfn "FSharpEmbedResourceText: %s" message
-            let justfilename = Path.GetFileNameWithoutExtension(filename) // .txt
-            if justfilename |> Seq.exists (System.Char.IsLetterOrDigit >> not) then
-                Err(filename, 0, sprintf "The filename '%s' is not allowed; only letters and digits can be used, as the filename also becomes the namespace for the SR class" justfilename)
-            let outFilename = Path.Combine(_outputPath, justfilename + ".fs")
-            let outXmlFilename = Path.Combine(_outputPath, justfilename + ".resx")
+          let printMessage message = printfn "FSharpEmbedResourceText: %s" message
+          let justfilename = Path.GetFileNameWithoutExtension(filename) // .txt
+          if justfilename |> Seq.exists (System.Char.IsLetterOrDigit >> not) then
+              Err(filename, 0, sprintf "The filename '%s' is not allowed; only letters and digits can be used, as the filename also becomes the namespace for the SR class" justfilename)
+          let outFilename = Path.Combine(_outputPath, justfilename + ".fs")
+          let outXmlFilename = Path.Combine(_outputPath, justfilename + ".resx")
+
+          if File.Exists(outFilename) && 
+               File.Exists(outXmlFilename) && 
+               File.Exists(filename) && 
+               File.GetLastWriteTime(filename) <= File.GetLastWriteTime(outFilename) &&
+               File.GetLastWriteTime(filename) <= File.GetLastWriteTime(outXmlFilename) then 
+            printMessage (sprintf "Skipping generation of %s and %s since up-to-date" outFilename outXmlFilename)
+            Some (outFilename, outXmlFilename)
+          else
 
             printMessage (sprintf "Reading %s" filename)
             let lines = File.ReadAllLines(filename) 
@@ -472,27 +481,28 @@ open Printf
             with get() = _hostObject
              and set(value) = _hostObject <- value
         member this.Execute() =
-            let sourceItem (source:string) (originalItem:string) =
-                let item = TaskItem(source)
-                item.SetMetadata("AutoGen", "true")
-                item.SetMetadata("DesignTime", "true")
-                item.SetMetadata("DependentUpon", originalItem)
-                item :> ITaskItem
-            let resxItem (resx:string) =
-                let item = TaskItem(resx)
-                item.SetMetadata("ManifestResourceName", Path.GetFileNameWithoutExtension(resx))
-                item :> ITaskItem
-            let generatedFiles, generatedResult =
+
+            let generatedFiles =
                 this.EmbeddedText
-                |> Array.fold (fun (resultList, aggregateResult) item ->
-                    match generateResxAndSource item.ItemSpec with
-                    | Some (source, resx) -> (((source, resx) :: resultList), aggregateResult)
-                    | None -> (resultList, false)
-                ) ([], true)
+                |> Array.choose (fun item -> generateResxAndSource item.ItemSpec)
+
             let generatedSource, generatedResx =
-                generatedFiles
-                |> List.map (fun (source, resx) -> (sourceItem source resx, resxItem resx))
-                |> List.fold (fun (sources, resxs) (source, resx) -> (source :: sources, resx:: resxs)) ([], [])
-            _generatedSource <- generatedSource |> List.rev |> List.toArray
-            _generatedResx <- generatedResx |> List.rev |> List.toArray
+                [| for (source, resx) in generatedFiles do
+                    let sourceItem =
+                        let item = TaskItem(source)
+                        item.SetMetadata("AutoGen", "true")
+                        item.SetMetadata("DesignTime", "true")
+                        item.SetMetadata("DependentUpon", resx)
+                        item :> ITaskItem
+                    let resxItem =
+                        let item = TaskItem(resx)
+                        item.SetMetadata("ManifestResourceName", Path.GetFileNameWithoutExtension(resx))
+                        item :> ITaskItem
+                    yield (sourceItem, resxItem) |]
+                 |> Array.unzip
+
+            let generatedResult = (generatedFiles.Length = this.EmbeddedText.Length)
+
+            _generatedSource <- generatedSource
+            _generatedResx <- generatedResx
             generatedResult

--- a/vsintegration/src/FSharp.ProjectSystem.FSharp/ProjectSystem.fsproj
+++ b/vsintegration/src/FSharp.ProjectSystem.FSharp/ProjectSystem.fsproj
@@ -42,7 +42,7 @@
   <PropertyGroup>
     <Win32Resource>$(IntermediateOutputPath)\ProjectResources.rc.res</Win32Resource>
   </PropertyGroup>
-  <Target Name="BeforeBuild">
+  <Target Name="BeforeBuild" Condition="!Exists('$(IntermediateOutputPath)$(RCResourceFile).res')">
     <Exec Command="&quot;$(ProgramFiles)\Windows Kits\8.1\bin\x86\rc.exe&quot; /fo $(IntermediateOutputPath)$(RCResourceFile).res $(RCResourceFile)" Condition="Exists('$(ProgramFiles)\Windows Kits\8.1\bin\x86\rc.exe')" />
     <Exec Command="&quot;$(ProgramFiles)\Windows Kits\10\bin\x86\rc.exe&quot; /fo $(IntermediateOutputPath)$(RCResourceFile).res $(RCResourceFile)" Condition="Exists('$(ProgramFiles)\Windows Kits\10\bin\x86\rc.exe')" />
     <Exec Command="&quot;$(ProgramFiles)\Windows Kits\10\bin\10.0.15063.0\x86\rc.exe&quot; /fo $(IntermediateOutputPath)$(RCResourceFile).res $(RCResourceFile)" Condition="Exists('$(ProgramFiles)\Windows Kits\10\bin\10.0.15063.0\x86\rc.exe')" />

